### PR TITLE
feat(products):  added POST /products route

### LINKED ISSUE #22: Add endpoint `POST /products`
### Request

- `name` - Product Name
- `thumbnail` - Product Thumbnail (Photo ID)
- `short_description` - Short Product Description
- `description` - Product Description
- `likes` - Product Like Count
- `stock` - Product Stock Count
- `price` - Product price
- `variations` - Array of *product_variations_id*

### Response

- `success` - If inserting student success or not
- `message` - Message to the client
- `data` - *empty*

### PR DESCRIPTION
**Pull request by ampats04**
closes #22

### CODE REVIEW

File: src/db/models/product.ts [hunk 1]
```diff
@@ -117,6 +119,122 @@ class Product extends DatabaseModel {
...
+  /**
+   * Validate Product Data
+   * @param data Product Data
+   */
+
+  public static validate(data: any){
+    //check if name is empty
+    if (!data.name) return ["Name is required!", "name"];
+    //check if thumbnail is empty
+    if (data.thumbnail.trim().length === 0) return ["Thumbnail is required!", "thumbnail"];
```
Comment (mavyfaby) [inline src/db/models/product.ts]: Set thumbnail data type to int (Photo ID)

File: src/db/models/product.ts [hunk 2]
```diff
@@ -117,6 +119,122 @@ class Product extends DatabaseModel {
...
+    //check if Short Description is empty
+    if (!data.short_descprition) return ["Short Description is required!", "short_description"];
+    //Check if Short Description doesn't exceed 255 characters
+    if (data.short_descprition.length > 255) return ["Short Description must not exceed 255 characters!", "short_description"];
+    //Check if Likes is not less than 0
+    if (data.likes.length < 0 ) return ["Likes must not be below 0", "likes"];
+    //Check if stocks is not less than 0
+    if (data.stock.length < 0) return ["Stocks must not be below 0", "stock"];
```
Comment (mavyfaby) [inline src/db/models/product.ts]: Add variations also which is a custom type:

```ts
type ProductVariation {
    type: string; // Variation type e.g. ("color", "sizes")
    name: string; // Variation name e.g. ("Red", "Large")
    photoID: number; // Product variation photo ID
}
```

**Comment by mavyfaby** [comment]
With introduction of product variations, we need to also add a bridge table for `products` and `variations`.

**Review by mavyfaby — CHANGES_REQUESTED** [review]

**Comment by mavyfaby** [comment]
Re-opened issue #4 and #5 due to changes.

**Comment by mavyfaby** [comment]
Updates?

**Comment by ampats04** [comment]
I'll push the one I've made last week, still wasn't able to continue due to
High Fever and Extreme Dehydration

**Review by mavyfaby — APPROVED** [review]

**Pull request merged**